Adds tutorial from blog and adjusts Readme to accommodate new tutorial (#41)
Esri/Dejavu: README.md

    <img src="https://img.shields.io/badge/swift-5.9-orange">
</div>

> Dejavu is Esri’s open‑source, Swift‑native library that records real HTTP requests and replays them later, so your test suite runs fast and predictably. It plugs into URLSession via URLProtocol, stores request/response pairs in a SQLite cache, and plays them back on demand--all while fitting neatly into XCTest.

Reviewer: ```suggestion
Dejavu is Esri’s open‑source, Swift‑native library that records real HTTP requests and replays them later, so your test suite runs fast and predictably. It plugs into `URLSession` via `URLProtocol`, stores request/response pairs in a SQLite cache, and plays them back on demand--all while fitting neatly into XCTest.
```

> Dejavu is used by the ArcGIS Maps SDK for Swift team to help test the [ArcGIS Maps SDK for Swift](https://github.com/Esri/arcgis-maps-sdk-swift).

Reviewer 2: This wording sounds a bit redundant. Maybe:

```suggestion
Dejavu is used by the team behind the [ArcGIS Maps SDK for Swift](https://github.com/Esri/arcgis-maps-sdk-swift).
```

> # Tutorial: Setting up and using Dejavu  

Reviewer: Was it necessary to switch from level two to level one headings? If not, please revert those changes. You could then create an issue and/or a PR for discussion of that particular change.

> # Tutorial: Setting up and using Dejavu  

Reviewer 2: Using heading level 1 causes this text to render larger than the `<h1>Dejavu</h1>` in the header, which causes the hierarchy to look off to me. I think sticking with heading level 2 and lower here and below would be better.

> The [Tutorial: Setting up and using Dejavu](./Tutorial) steps you through setting up Dejavu, configuring a session, and using Dejavu.

Reviewer: ```suggestion
# Tutorial: Setting up and using Dejavu

The [Tutorial: Setting up and using Dejavu](AdditionalDocumentation/Tutorial.md) steps you through setting up Dejavu, configuring a session, and using Dejavu.
```

Author: Resolved locally

## Example
# Tutorial: Setting up and using Dejavu  
The [Tutorial: Setting up and using Dejavu](./Tutorial) steps you through setting up Dejavu, configuring a session, and using Dejavu.

> A full example of a mocked network test can be found [here](Examples/ExamplesTests/ExamplesTests.swift).

Reviewer 2: The benefit of having the example in a Swift file is that it can be easily opened, compiled and ran. If we point to code snippets in a markdown tutorial instead, it'll require readers to do a good amount of copy-pasting just to get a working example.

Maintenance is another consideration. With the Example project it can be easily opened to see any new warnings that might arise over time as Xcode evolves.

> * [Tutorial: Setting up and using Dejavu](./Tutorial)

Reviewer: ```suggestion
* [Tutorial: Setting up and using Dejavu](AdditionalDocumentation/Tutorial.md)
```

Author: Resolved locally

* [Record requests, play back requests, and more](record-requests-play-back-requests-and-more)
* [A full example of a mocked network test](Examples/ExamplesTests/ExamplesTests.swift)

> * One time setup: [Configure a location to store mocked data](./AdditionalDocumentation/MockedDataSetupInstructions.md)

Reviewer: ```suggestion
* One time setup: [Configure a location to store mocked data](AdditionalDocumentation/MockedDataSetupInstructions.md)
```

* Optional: Use custom network interceptors and observers

## Requirements

* Swift 5.9 / Xcode 15.0 (or newer)

> * iOS 15.0, Mac Catalyst 15.0 (minimum deployment targets)

Reviewer: According to the package manifest, the current minimum is 16.0. Might as well update this now.

```suggestion
* iOS 16.0, Mac Catalyst 16.0 (minimum deployment targets)
```

Author: Resolved locally

- `playback` - Intercepts requests and gets the responses from the cache.

# Issues

Find a bug or want to request a new feature?  Please let us know by [submitting an issue](https://github.com/ArcGIS/Dejavu/issues/new).

# Contributing

Esri welcomes contributions from anyone and everyone. Please see our [guidelines for contributing](https://github.com/esri/contributing).

# Licensing

> Copyright 2023 Esri

Reviewer: ```suggestion
# Licensing

Copyright 2023 Esri
```

limitations under the License.

A copy of the license is available in the repository's [LICENSE.txt](LICENSE.txt) file.

---

> **TODO** move this section's intro text to the dejavuconfiguration section after clarifying what is needed when using defaults vs custom interceptors and observers (the snippet below already appears in the tutorial)

Reviewer 2: I think any TODOs would need to be resolved before merging this.
